src/gen_smtp_server_session.erl

```diff
@@ -972,12 +979,21 @@ check_bare_crlf(Binary, _Prev, Op, Offset) ->
 			end
 	end.
 
-send(#state{transport = Transport, socket = Sock}, Data) ->
-    %% TODO: handle send errors
-    Transport:send(Sock, Data).
+send(#state{transport = Transport, socket = Sock} = St, Data) ->
+    case Transport:send(Sock, Data) of
+		ok -> ok;
+		{error, Err} ->
+			St1 = handle_error(send_error, Err, St),
+			throw({stop, {send_error, Err}, St1})
+	end.
 
-setopts(#state{transport = Transport, socket = Sock}, Opts) ->
-    ok = Transport:setopts(Sock, Opts).
+setopts(#state{transport = Transport, socket = Sock} = St, Opts) ->
+    case Transport:setopts(Sock, Opts) of
+		ok -> ok;
+		{error, Err} ->
+			St1 = handle_error(setopts_error, Err, St),
+			throw({stop, {setopts_error, Err}, St1})
+	end.
```

Comment:
it used to fail with smth like
```
[2020-10-29T05:16:17.107810+01:00 error <0.27720.227> proc_lib:crash_report/4:508]
crasher:
 initial call: gen_smtp_server_session:ranch_init/1,
 pid: <0.27720.227>,
 registered_name: [],
error: {{badmatch,{error,closed}},[{gen_smtp_server_session,setopts,2,[{file,"<..>/_build/default/lib/gen_smtp/src/gen_smtp_server_session.erl"},{line,993}]},
{gen_smtp_server_session,handle_info,2,[{file,"<..>/_build/default/lib/gen_smtp/src/gen_smtp_server_session.erl"},{line,283}]},
{gen_server,try_dispatch,4,[{file,"gen_server.erl"},{line,637}]},
{gen_server,handle_msg,6,[{file,"gen_server.erl"},{line,711}]},
{proc_lib,init_p_do_apply,3,[{file,"proc_lib.erl"},{line,249}]}]},
ancestors: [<0.1007.0>,<0.1006.0>,ranch_sup,<0.919.0>],
message_queue_len: 1,
messages: [{ssl_closed,{sslsocket,{gen_tcp,#Port<0.6554831>,tls_connection,undefined},[<0.21175.227>,<0.18409.227>]}}],
links: [<0.1007.0>],
 dictionary: [],
 trap_exit: false,
 status: running,
 heap_size: 121536,
 stack_size: 27,
 reductions: 300084;
 neighbours:
```
Now `handle_error` callback will be called first and it can return `{stop, normal, ...}` to avoid SASL crash logs